cmake/onnxruntime_training.cmake

```diff
@@ -230,4 +230,31 @@ if (onnxruntime_BUILD_UNIT_TESTS)
   target_link_libraries(onnxruntime_training_gpt2 PRIVATE onnxruntime_training_runner onnxruntime_training ${ONNXRUNTIME_LIBS} ${onnxruntime_EXTERNAL_LIBRARIES})
   set_target_properties(onnxruntime_training_gpt2 PROPERTIES FOLDER "ONNXRuntimeTest")
 
+  # Training API Tests
+  # Currently disable it for internval development usage.
+  if (onnxruntime_ENABLE_TRAINING_API)
+    file(GLOB_RECURSE training_api_test_runner_src
+        "${ORTTRAINING_SOURCE_DIR}/test/training_api/*.h"
+        "${ORTTRAINING_SOURCE_DIR}/test/training_api/*.cc"
+    )
+    onnxruntime_add_executable(onnxruntime_training_api_test_runner ${training_api_test_runner_src})
+
+    if(UNIX AND NOT APPLE)
+      if (HAS_NO_MAYBE_UNINITIALIZED)
+        target_compile_options(onnxruntime_training_api_test_runner PUBLIC "-Wno-maybe-uninitialized")
+      endif()
+    endif()
+
+    onnxruntime_add_include_to_target(onnxruntime_training_api_test_runner onnxruntime_common onnx onnx_proto ${PROTOBUF_LIB} onnxruntime_training flatbuffers)
+    target_include_directories(onnxruntime_training_api_test_runner PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${ONNXRUNTIME_ROOT} ${ORTTRAINING_ROOT} ${MPI_CXX_INCLUDE_DIRS} ${eigen_INCLUDE_DIRS} ${CXXOPTS} ${extra_includes} ${onnxruntime_graph_header} ${onnxruntime_exec_src_dir} ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR}/onnx onnxruntime_training_runner)
+
+    # ROCM provider sources are generated, need to add include directory for generated headers
+    if (onnxruntime_USE_ROCM)
```

Comment:
do we need this? At the moment target for on device training is nvidia gpu and cpu.

Reply:
I thought we should always keep CUDA and ROCM in parity, because we have many AMD GPUs for internal customers. 

But to reduce some efforts, it's fine to remove it now.